tests/embed/test_gripnet.py

```diff
@@ -35,3 +36,113 @@ def test_gripnet_encoder(pose_data):
+
+# determine the supervertices among them
+edge_index = torch.tensor([[0, 1, 2, 3], [1, 1, 3, 3]])
+
+superedge1 = SuperEdge(supervertex1.name, supervertex3.name, edge_index)
+superedge2 = SuperEdge(supervertex2.name, supervertex3.name, edge_index)
+
+# create a supergraph
+supergraph = SuperGraph([supervertex1, supervertex2, supervertex3], [superedge1, superedge2])
+
+
+def test_gripnet_internal_module1():
+    """GripNet Internal Module Test for start supervertex"""
+
+    setting1 = SuperVertexParaSetting("start_supervertex", 20, [10, 10])
+    inter_module1 = GripNetInternalModule(
+        supervertex1.num_node_feat, supervertex1.num_edge_type, supervertex1.if_start_supervertex, setting1
+    )
+
+    x = torch.randn(20, 20)
+    y = inter_module1(x, supervertex3.edge_index)
+
+    assert y.shape[1] == inter_module1.out_channels
+
+
+def test_gripnet_internal_module2():
+    """GripNet Internal Module Test for end supervertex"""
+
+    setting2 = SuperVertexParaSetting("task_supervertex", 20, [15, 10], exter_agg_dim={"1": 20, "2": 20}, mode="cat")
+    inter_module2 = GripNetInternalModule(
+        supervertex3.num_node_feat, supervertex3.num_edge_type, supervertex3.if_start_supervertex, setting2
+    )
+
+    x = torch.randn(20, 20)
+    x1 = torch.matmul(x, inter_module2.embedding)
+    x2 = torch.randn(20, 40)
+    xx = torch.cat((x1, x2), dim=1)
+
+    range_list = torch.LongTensor([[0, 2], [2, 4]])
+    edge_weight = torch.randn(4)
+
+    y = inter_module2(xx, supervertex3.edge_index, supervertex3.edge_type, range_list, edge_weight)
+
+    assert y.shape[1] == inter_module2.out_channels
+
+
+def test_gripnet_external_module():
+    """GripNet External Module Test"""
+
+    exter_module = GripNetExternalModule(8, 7, 5)
+    x = torch.randn((4, 8))
+    edge_index = torch.tensor([[0, 1, 2, 3], [1, 4, 3, 4]])
+    y = exter_module(x, edge_index)
+
+    assert y.shape[0] == 5
+    assert y.shape[1] == 7
+
+
+def test_gripnet_cat():
+    """GripNet Cat Test"""
+    setting1 = SuperVertexParaSetting("1", 20, [10, 10])
+    setting2 = SuperVertexParaSetting("2", 20, [10, 10])
+    setting3 = SuperVertexParaSetting("3", 11, [15, 10], exter_agg_dim={"1": 12, "2": 13}, mode="cat")
+
+    supergraph.set_supergraph_para_setting([setting1, setting2, setting3])
+
+    gripnet = GripNet(supergraph)
+
+    assert (
+        gripnet.supervertex_module_dict["3"][-1].inter_agg_layers[0].in_channels == 11 + 12 + 13
+    ), "ValueError: invalid exter_agg_dim settings in the task vertex."
+
+    y = gripnet()
+
+    error_message = "ValueError: dimension mismatch in the task vertex"
+
+    assert gripnet.out_embed_dict["1"].shape[1] == 20 + 10 + 10, error_message
+    assert gripnet.out_embed_dict["2"].shape[1] == 20 + 10 + 10, error_message
+    assert y.shape[1] == 11 + 12 + 13 + 15 + 10, error_message
+
+    # general tests
+    assert gripnet.__repr__() is not None
+
+
+def test_gripnet_add():
+    """GripNet Add Test"""
```

Comment:
see whether `test_gripnet_add()` can be merged with `test_gripnet_cat` above since they share the same logic and flow,  i.e., add and cat are just two values of the same parameter so you may use parameterize of pytest. 
Example: https://github.com/pykale/pykale/blob/main/tests/embed/test_factorization.py#L87, where multiple kernels and aug options are tested.
Doc: https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
Thx. Updated with `pytest.mark.parametrize` in [19889fd](https://github.com/pykale/pykale/pull/338/commits/19889fd45cbd7c3d607193422377b4df978a5c7c).